src/mongo_logger.ts

```diff
@@ -415,10 +417,15 @@ export function stringifyWithMaxLen(
 ): string {
   let strToTruncate = '';
 
-  if (typeof value === 'function') {
-    strToTruncate = value.toString();
-  } else {
-    strToTruncate = EJSON.stringify(value, options);
+  try {
+    strToTruncate =
+      typeof value !== 'function'
+        ? EJSON.stringify(value, options)
+        : value.name !== ''
+        ? value.name
+        : 'anonymous function';
+  } catch (e) {
+    strToTruncate = `... ESJON failed : Error ${e.message}`;
```

Comment:
There's a typo here (`EJSON`) but it might be worth wording this a bit more explicitly anyway:

```suggestion
    strToTruncate = `Extended JSON serialization failed with: ${e.message}`;
```

Reply:
Sounds good to me!